fix: update gVisor documentation to use router service and gateway (#169)
kubernetes-sigs/agent-sandbox: examples/vscode-sandbox/README.md

$ kubectl get pods -o jsonpath=$'{range .items[*]}{.metadata.name}: {.spec.runtimeClassName}\n{end}'
```

## Accessing VSCode

To access the VSCode instance securely and reliably, use the Sandbox Router.

> ### Option 1: Production (via Gateway)

Reviewer: Given that this adds an additional step to deploy the Gateway and Router, and the use of gVisor / Kata is optional in this example, we should keep the original simple approach for users not using those runtimes.

Reviewer: Actually, the original doc is missing a critical step to wait for the VSCode server to be actually running (pod running isn't enough). 
The pod log should have something like this:
```
+ /usr/bin/code-server --auth=none --bind-addr=0.0.0.0:13337                                                                    
[2025-11-19T01:09:55.176Z] info  Wrote default config file to /root/.config/code-server/config.yaml                             
[2025-11-19T01:09:55.627Z] info  code-server 4.105.1 811ec6c1d60add2eb92446161ca812828fdbaa7f                                   
[2025-11-19T01:09:55.629Z] info  Using user-data-dir /root/.local/share/code-server                                             
[2025-11-19T01:09:55.653Z] info  Using config file /root/.config/code-server/config.yaml                                        
[2025-11-19T01:09:55.653Z] info  HTTP server listening on http://0.0.0.0:13337/                                                 
[2025-11-19T01:09:55.653Z] info    - Authentication is disabled                                                                 
[2025-11-19T01:09:55.654Z] info    - Not serving HTTPS                                                                          
[2025-11-19T01:09:55.655Z] info  Session server listening on /root/.local/share/code-server/code-server-ipc.sock
```

Before users can actually access it. 

@barney-s to confirm

Author: Added this explanation as well.

> If you have deployed the Gateway **and** Router (see `clients/python/agentic-sandbox-client/sandbox_router`), access VSCode via the external Gateway IP.

Reviewer: Add a link to the sandbox_router folder for easier reference

1. Get the Gateway IP:
```bash
export GATEWAY_IP=$(kubectl get gateway external-http-gateway -n default -o jsonpath='{.status.addresses[0].value}')
echo "Gateway IP: $GATEWAY_IP"
```

2. **Connect:** You must inject the routing headers. Use `curl` to test:
```bash
# Replace <GATEWAY_IP> with the actual IP

> curl -v -H "X-Sandbox-ID: sandbox-example" \

Reviewer 2: has this been tested ?  accessing the vscode UI via router ?

Author: Yeah i can add instructions for the UI as well 
<img width="3424" height="1594" alt="image" src="https://github.com/user-attachments/assets/604c3a5f-c9ad-4653-8933-96033a45e050" />

        -H "X-Sandbox-Port: 13337" \
        http://$GATEWAY_IP
```

### Option 2: Development (via Router Tunnel)

> For local development, port-forward to the **Router Service** (do not port-forward to the pod directly, as it may fail with secure runtimes like gVisor).

Reviewer: Router is a prerequisite for option 2 as well, right? So the router installation step needs to be called out in option 2 or move up to above both options. 

Also suggest changing the vague "may fail" text here:

```suggestion
For local development, port-forward to the **Router Service** (do not port-forward to the pod directly, as it's not compatible with secure runtimes like gVisor).
```
